vscode/src/testExplorer.ts

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// This file uses the VS Code Test Explorer API (https://code.visualstudio.com/docs/editor/testing)
+
+import * as vscode from "vscode";
+import { loadProject } from "./projectSystem";
+import {
+  getCompilerWorker,
+  ICompilerWorker,
+  IProjectConfig,
+  log,
+  ProgramConfig,
+  QscEventTarget,
+} from "qsharp-lang";
+import { getActiveQSharpDocumentUri } from "./programConfig";
+import { IProgramConfig } from "../../npm/qsharp/lib/web/qsc_wasm";
+import { getTarget } from "./config";
+import { toVsCodeRange } from "./common";
+
+function localGetCompilerWorker(
+  context: vscode.ExtensionContext,
+): ICompilerWorker {
+  const compilerWorkerScriptPath = vscode.Uri.joinPath(
+    context.extensionUri,
+    "./out/compilerWorker.js",
+  ).toString();
+  const worker = getCompilerWorker(compilerWorkerScriptPath);
+  return worker;
```

Comment:
Bah, these two lines are duplicated in so many places (not your fault). If you're feeling boy scout-y, I'd love to have it refactored into a common utility.